src/mappings/serviceAgreement.ts

```diff
@@ -15,29 +15,32 @@ export async function handleServiceAgreementCreated(
   logger.info('handleClosedServiceAgreementCreated');
   assert(event.args, 'No event args');
 
-  const saContract = IServiceAgreement__factory.connect(
-    event.args.serviceAgreement,
+  const eventServiceAgreementId = event.args.serviceAgreementId;
+
+  const agreementRegistry = IServiceAgreementRegistry__factory.connect(
+    eventServiceAgreementId.toString(),
     new FrontierEthProvider()
   );
 
-  const [period, value] = await Promise.all([
-    saContract.period(),
-    saContract.value(),
-  ]);
+  const agreement = await agreementRegistry.getClosedServiceAgreement(
+    eventServiceAgreementId
```

Comment:
`eventServiceAgreementId` should be `agreementRegistry` contract address

Reply:
` const eventServiceAgreementId = event.args.serviceAgreementId.toString();

  const agreementRegistry = IServiceAgreementRegistry__factory.connect(
    SA_REGISTRY_ADDRESS,
    new FrontierEthProvider()
  );

  const agreement = await agreementRegistry.getClosedServiceAgreement(
    eventServiceAgreementId
  );`

Reply:
```
 const eventServiceAgreementId = event.args.serviceAgreementId.toString();

  const agreementRegistry = IServiceAgreementRegistry__factory.connect(
    SA_REGISTRY_ADDRESS,
    new FrontierEthProvider()
  );

  const agreement = await agreementRegistry.getClosedServiceAgreement(
    eventServiceAgreementId
  );

Reply:
You mean it should be like this right?